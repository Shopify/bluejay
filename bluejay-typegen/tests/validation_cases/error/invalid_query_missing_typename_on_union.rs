#[bluejay_typegen::typegen([
    type Query {
        foo: Foo
    }

    type Bar {
        bar: String
    }

    type Baz {
        baz: String
    }

    union Foo = Bar | Baz
])]
mod schema {
    #[query([
        {
            foo {
                ...on Bar { bar }
                ...on Baz { baz }
            }
        }
    ])]
    mod query {}
}

fn main() {}
