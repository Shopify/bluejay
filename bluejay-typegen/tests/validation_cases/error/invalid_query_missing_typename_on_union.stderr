error: Error: bluejay_typegen requires unaliased selection of `__typename` on union types to properly deserialize, and for that to be the first in the selection set
          ╭─[<unknown>:1:7]
          │
        1 │ { foo { ...on Bar { bar } ...on Baz { baz } } }
          │       ───────────────────┬───────────────────
          │                          ╰───────────────────── Selection set does not contain an unaliased `__typename` selection as the first selection
       ───╯
  --> tests/validation_cases/error/invalid_query_missing_typename_on_union.rs:17:13
   |
17 |       #[query([
   |  _____________^
18 | |         {
19 | |             foo {
20 | |                 ...on Bar { bar }
...  |
23 | |         }
24 | |     ])]
   | |_____^
